ddl/index.go

```diff
@@ -626,6 +626,12 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 		var reorgTp model.ReorgType
 		reorgTp, err = pickBackfillType(w.ctx, job, indexInfo.Unique, d)
 		if err != nil {
+			if !errorIsRetryable(err, job) {
```

Comment:
Could we cancel it directly which needn't do `convertAddIdxJob2RollbackJob`(Here the job is StateNone) ?